File: tasks/2024/round4_pp/plusz.py
#!/usr/bin/env python3
# @check-accepted: examples small no-limits
#

V,A,B = list(map(int, input().strip().split()))
F,C,D = list(map(int, input().strip().split()))

if A+B==2*F and C+D==2*V:
    if B-A==D-C:
        print('IGEN')
    else:
        print('NEM')
else:
    print('NEM')

File: tasks/2024/round4_pp/plusz.cpp
// @check-accepted: examples small no-limits
// insert brief description of the solution here

#include <fstream>
#include <iostream>
#include <vector>

using namespace std;

int main() {

  int V,A,B,F,C,D;
  cin >> V >> A >> B;
  cin >> F >> C >> D;

  if (A+B==2*F and C+D==2*V) {
    if (B-A==D-C) {
      cout << "IGEN" << endl;
    } else {
      cout << "NEM" << endl;
    }
  } else {
    cout << "NEM" << endl;
  }
    
  return 0;
}
